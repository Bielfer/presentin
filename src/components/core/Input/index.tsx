import { FC, InputHTMLAttributes } from 'react';
import InputLayout from '../InputLayout';

interface Props extends InputHTMLAttributes<HTMLInputElement> {
  label?: string;
  password?: boolean;
  error?: string;
  hint?: string;
}

const Input: FC<Props> = ({
  label,
  placeholder,
  password,
  className,
  type,
  error,
  name,
  hint,
  ...props
}) => (
  <InputLayout
    name={name}
    className={className}
    error={error}
    hint={hint}
    label={label}
    shadow
  >
    <input
      className={
        !error
          ? 'shadow-sm focus:ring-blue-500 focus:border-blue-500 block w-full sm:text-sm border-gray-300 rounded-md'
          : 'block w-full pr-10 border-red-300 text-red-900 placeholder-red-300 focus:outline-none focus:ring-red-500 focus:border-red-500 sm:text-sm rounded-md'
      }
      placeholder={placeholder}
      type={password ? 'password' : type ?? 'text'}
      id={name}
      {...props}
    />
  </InputLayout>
);

export default Input;
