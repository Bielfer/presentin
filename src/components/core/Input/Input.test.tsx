import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Input from '.';

describe('Input', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Input />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should be able to type', async () => {
    render(<Input />);

    const RenderedInput = screen.getByRole('textbox');

    await waitFor(() => userEvent.type(RenderedInput, 'test'));

    expect(RenderedInput).toHaveValue('test');
  });

  it('should have a default type of text', () => {
    render(<Input />);

    const RenderedInput = screen.getByRole('textbox');

    expect(RenderedInput).toHaveAttribute('type', 'text');
  });
});
