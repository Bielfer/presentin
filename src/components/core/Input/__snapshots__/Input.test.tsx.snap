// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="mt-1 relative rounded-md shadow-sm"
    >
      <input
        class="shadow-sm focus:ring-blue-500 focus:border-blue-500 block w-full sm:text-sm border-gray-300 rounded-md"
        type="text"
      />
    </div>
  </div>
</DocumentFragment>
`;
